### PR TITLE
add doxygen documentation

### PR DESCRIPTION
**Pull request by mayank-sirotiya-imgtec**
Signed-off-by: Mayank Sirotiya <Mayank.Sirotiya@imgtec.com>

### CODE REVIEW

File: docs/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Add docs targets
+#################
+FIND_PACKAGE(Doxygen)
+IF(DOXYGEN_FOUND)
+  OPTION(DOCS_INTERNAL "enable internal docs generation" OFF)
```
Comment (abhijit-mahajani-imgtec) [inline docs/CMakeLists.txt]: why do we need internal docs?

**Comment by nikhil-zinjurde-imgtec** [comment]
Approved

**Comment by rahul-daga-imgtec** [comment]
Approved & Verified

**Comment by abhijit-mahajani-imgtec** [comment]
Approved and Verified

**Pull request merged**